support-lambdas/bigquery-acquisitions-publisher/typescript/acquisitions.ts

```diff
@@ -0,0 +1,137 @@
+	browserId: z.string().nullable(),
+	identityId: z.string(),
+	pageViewId: z.string(),
+	referrerPageViewId: z.string().nullable(),
+	promoCode: z.string().nullable(),
+	queryParameters: z.object({ name: z.string(), value: z.string() }).array(),
+	reusedExistingPaymentMethod: z.boolean(),
+	acquisitionType: z.string(),
+	readerType: z.string(),
+	zuoraSubscriptionNumber: z.string().nullable(),
+	contributionId: z.string().nullable(),
+	paymentId: z.string().nullable(),
+	product: z.enum(ProductTypeSchema),
+	amount: z.number().nullable(),
+	currency: z.enum(IsoCurrencySchema),
+	source: z.string().nullable(),
+	platform: z.string().nullable(),
+	labels: z.string().array(),
+});
+export type AcquisitionProduct = z.infer<typeof AcquisitionProductSchema>;
+export const AcquisitionProductEventSchema = z.object({
+	detail: AcquisitionProductSchema,
+});
+export type AcquisitionProductEvent = z.infer<
+	typeof AcquisitionProductEventSchema
+>;
+
+export type AcquisitionProductBigQueryType = {
+	event_timestamp: string;
+	product: ProductType;
+	amount: number | null;
+	currency: IsoCurrencyType;
+	country_code: IsoCountryType;
+	component_id: string | null;
+	component_type: string | null;
+	campaign_codes: [string] | [];
+	referrer_url: string | null;
+	ab_tests: { name: string; variant: string }[];
+	payment_frequency: ContributionType;
+	payment_provider: PaymentMethodType;
+	print_options: { product: string; delivery_country_code: string }[] | null;
+	browser_id: string | null;
+	identity_id: string;
+	page_view_id: string;
+	referrer_page_view_id: string | null;
+	promo_code: string | null;
+	query_parameters: { key: string; value: string }[];
+	reused_existing_payment_method: boolean;
+	acquisition_type: string;
+	reader_type: string;
+	zuora_subscription_number: string | null;
+	contribution_id: string | null;
+	payment_id: string | null;
+	platform: string | null;
+	labels: string[];
+};
+
+const mapPlatformName = (name: string): string => {
+	switch (name.toLowerCase()) {
+		case 'iosnativeapp':
+			return 'IOS_NATIVE_APP';
+		case 'androidnativeapp':
+			return 'ANDROID_NATIVE_APP';
+		default:
+			return name;
+	}
+};
+
+export const transformAcquisitionProductForBigQuery = (
+	acquisitionProduct: AcquisitionProduct,
+): AcquisitionProductBigQueryType => {
+	return {
+		event_timestamp: acquisitionProduct.eventTimeStamp,
+		product: acquisitionProduct.product,
+		amount: acquisitionProduct.amount,
+		currency: acquisitionProduct.currency,
+		country_code: acquisitionProduct.country,
+		component_id: acquisitionProduct.componentId,
+		component_type: acquisitionProduct.componentType,
+		campaign_codes: acquisitionProduct.campaignCode
+			? [acquisitionProduct.campaignCode]
+			: [],
+		referrer_url: acquisitionProduct.referrerUrl,
+		ab_tests: acquisitionProduct.abTests,
+		payment_frequency: acquisitionProduct.paymentFrequency,
+		payment_provider: acquisitionProduct.paymentProvider,
+		print_options: acquisitionProduct.printOptions,
```

Comment:
I think you will need to map acquisitionProduct.printOptions to convert the `delivery_country_code` field